scripts/gen_layer.py

```diff
@@ -0,0 +1,3601 @@
+#include <valijson/schema.hpp>
+#include <valijson/schema_parser.hpp>
+#include <valijson/validation_results.hpp>
+#include <valijson/validator.hpp>
+
+#include <json/json.h>  // https://github.com/open-source-parsers/jsoncpp
+
+#include "vulkan/vk_layer.h"
+#include "vulkan/vulkan_beta.h"
+#include <vk_layer_config.h>
+#include "vk_layer_table.h"
+#include "vk_layer_settings.h"
+#include "profiles.h"
+
+using valijson::Schema;
+using valijson::SchemaParser;
+using valijson::ValidationResults;
+using valijson::Validator;
+using valijson::adapters::JsonCppAdapter;
+'''
+
+BEGIN_NAMESPACE = '''
+namespace {
+'''
+
+END_NAMESPACE = '''
+}  // anonymous namespace
+'''
+
+GLOBAL_CONSTANTS = '''
+// Global constants //////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// For new features/functionality, increment the minor level and reset patch level to zero.
+// For any changes, at least increment the patch level.  See https://semver.org/
+// When updating the version, be sure to make corresponding changes to the layer manifest file at
+// layer/VkLayer_khronos_profiles.json.in
+
+const uint32_t kVersionProfilesMajor = 1;
+const uint32_t kVersionProfilesMinor = 0;
+const uint32_t kVersionProfilesPatch = 0;
+const uint32_t kVersionProfilesImplementation =
+    VK_MAKE_VERSION(kVersionProfilesMajor, kVersionProfilesMinor, kVersionProfilesPatch);
+
+static const char *SCHEMA_URI_BASE = "https://schema.khronos.org/vulkan/profiles-";
+
+// Properties of this layer:
+const VkLayerProperties kLayerProperties[] = {{
+    "VK_LAYER_KHRONOS_profiles",     // layerName
+    VK_MAKE_VERSION(1, 0, 68),       // specVersion (clamped to final 1.0 spec version)
+    kVersionProfilesImplementation,  // implementationVersion
+    "Khronos Profiles layer"         // description
+}};
+const uint32_t kLayerPropertiesCount = (sizeof(kLayerProperties) / sizeof(kLayerProperties[0]));
+
+// Name of this layer:
+const char *kOurLayerName = kLayerProperties[0].layerName;
+
+// Instance extensions that this layer provides:
+const std::array<VkExtensionProperties, 0> kInstanceExtensionProperties = {};
+const uint32_t kInstanceExtensionPropertiesCount = static_cast<uint32_t>(kInstanceExtensionProperties.size());
+
+// Device extensions that this layer provides:
+const std::array<VkExtensionProperties, 2> kDeviceExtensionProperties = {
+    {{VK_EXT_TOOLING_INFO_EXTENSION_NAME, VK_EXT_TOOLING_INFO_SPEC_VERSION},
+     {VK_KHR_PORTABILITY_SUBSET_EXTENSION_NAME, VK_KHR_PORTABILITY_SUBSET_SPEC_VERSION}}};
+const uint32_t kDeviceExtensionPropertiesCount = static_cast<uint32_t>(kDeviceExtensionProperties.size());
+'''
+
+GLOBAL_VARS = '''
+uint32_t requested_version = 0;
+bool device_has_astc_hdr = false;
+bool device_has_astc = false;
+bool device_has_etc2 = false;
+bool device_has_bc = false;
+
+FILE *profiles_log_file = nullptr;
+
+static VkProfileLayerSettingsEXT *layer_settings;
```

Comment:
@ziga-lunarg, the layer tests changes were pushed this morning which included a small change to `profiles.cpp`, which will need to be updated here as well -- sorry!

Reply:
No problem, this is still WIP, I will compare everything to master before merging.